src/coffee/compose.coffee

```diff
@@ -21,10 +21,17 @@ module.exports = (config) ->
     path = path[1...] if path[0] is '/'
     resolvep root, path
 
-  _addExtraLabels: addExtraLabels = (service) ->
-    labels = _.extend {}, service.labels,
+  _addExtraLabels: addExtraLabels = (serviceName, service, instance) ->
+    labels = _.extend {}, service.labels, (service.deploy?.labels or {}),
```

Comment:
No need to protect against undefined `service.deploy.labels`, I believe `_.extend` handles it fine.